src/main/webapp/app/shared/modal/AddMutationModal.tsx

```diff
@@ -80,507 +82,747 @@ function AddMutationModalContent({
+      ...(await Promise.all(alterationPromises)).map((alt, index) =>
+        convertEntityStatusAlterationToAlterationData(alt, newParsedAlteration[index].alteration, [], newComment)
+      ),
+    ];
+
+    setTabStates(states => {
+      const newStates = _.cloneDeep(states);
+      newStates[tabIndex].excluding.splice(excludingIndex, 1, ...newAlterations);
+      return newStates;
     });
   }
 
-  const handleAlterationChange = useCallback(
-    _.debounce(async (newAlteration: string, uuid: string, excludingUuid?: string) => {
-      await fetchAlteration(newAlteration, uuid, excludingUuid);
+  const fetchExcludedAlterationDebounced = useCallback(
+    _.debounce((newAlteration: string, tabIndex: number, excludingIndex: number, alterationData: AlterationData[]) => {
+      fetchExcludedAlteration(newAlteration, tabIndex, excludingIndex, alterationData);
     }, 1000),
     []
   );
 
+  function handleAlterationChange(newValue: string, tabIndex: number, excludingIndex?: number, isDebounced = true) {
+    if (notNullOrUndefined(excludingIndex)) {
+      if (isDebounced) {
+        handleExcludingFieldChange(newValue, 'alteration', tabIndex, excludingIndex);
+        fetchExcludedAlterationDebounced(newValue, tabIndex, excludingIndex, tabStates);
+      } else {
+        fetchExcludedAlteration(newValue, tabIndex, excludingIndex, tabStates);
+      }
+    } else {
+      if (isDebounced) {
+        handleNormalAlterationChange(newValue, tabIndex);
+      } else {
+        fetchNormalAlteration(newValue, tabIndex, tabStates);
+      }
+    }
+  }
+
+  function handleNormalFieldChange(newValue: string, field: keyof AlterationData, tabIndex: number) {
+    setTabStates(states => {
+      const newStates = _.cloneDeep(states);
+      newStates[tabIndex][field as string] = newValue;
+      return newStates;
+    });
+  }
+
+  function handleExcludingFieldChange(newValue: string, field: keyof AlterationData, tabIndex: number, excludingIndex: number) {
+    setTabStates(states => {
+      const newStates = _.cloneDeep(states);
+      newStates[tabIndex].excluding[excludingIndex][field as string] = newValue;
+      return newStates;
+    });
+  }
+
+  function handleFieldChange(newValue: string, field: keyof AlterationData, tabIndex: number, excludingIndex?: number) {
+    notNullOrUndefined(excludingIndex)
+      ? handleExcludingFieldChange(newValue, field, tabIndex, excludingIndex)
+      : handleNormalFieldChange(newValue, field, tabIndex);
+  }
+
   const handleKeyDown: KeyboardEventHandler = async event => {
     if (!inputValue) return;
     if (event.key === 'Enter' || event.key === 'tab') {
-      await fetchAlteration(inputValue);
+      const newParsedAlteration = filterAlterationsAndNotify(parseAlterationName(inputValue), tabStates);
+      if (newParsedAlteration.length === 0) {
+        return;
+      }
+
+      const newComment = newParsedAlteration[0].comment;
+      const newVariantName = newParsedAlteration[0].name;
```

Comment:
Same here. Doesn't seem to me it's necessary to create two vars here.

Reply:
Correct, for same reasoning as above. Will change to use index.